lib/iris/fileformats/grib/__init__.py

```diff
@@ -911,9 +911,67 @@ def load_cubes(filenames, callback=None, auto_regularise=True):
     return iris.fileformats.rules.load_cubes(filenames, callback, grib_loader)
 
 
+def messages_from_filename(filename):
+    """
+    Return a generator of :class:`iris.fileformats.grib._message._GribMessage`
+    instances; one for each message in the supplied GRIB file.
+
+    Args:
+
+    * filename (string):
+        Name of the file to generate fields from.
+
+    """
+    return _GribMessage.messages_from_filename(filename)
+
+
+def load_pairs_from_fields(grib_messages):
+    """
+    Convert an iterable of GRIB messages into an iterable of
+    (Cube, Grib message) tuples.
+
+    Args:
+
+    * grib_messages:
+        An iterable of :class:`iris.fileformats.grib._message._GribMessage`.
+
+    Returns:
+        An iterable of tuples of (:class:`iris.cube.Cube`,
+        :class:`iris.fileformats.grib._message._GribMessage`).
+
+    This capability can be used to filter out fields before they are passed to
+    the load pipeline, and amend the cubes once they are created, using
+    GRIB metadata conditions.  Where the filtering
+    removes a significant number of fields, the speed up to load can be
+    significant::
+
+        filtered_messages = []
+        for message in iris.fileformats.grib.messages_from_filename(filename):
+            if message == 3:
```

Comment:
Should be like the others below... "message.productionStatusOfProcessedData", probably ??